mm2src/lp_ordermatch.rs

```diff
@@ -3549,6 +3579,125 @@ pub async fn order_status(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, St
+    pub was_taker: Option<bool>,
+    pub status: Option<String>,
+    #[serde(default)]
+    pub include_details: bool,
+}
+
+#[derive(Clone, Deserialize, Serialize)]
+#[serde(tag = "type", content = "order")]
+enum Order {
+    Maker(MakerOrder),
+    Taker(TakerOrder),
+}
+
+impl<'a> From<&'a Order> for OrderForRpc<'a> {
+    fn from(order: &'a Order) -> OrderForRpc {
+        match order {
+            Order::Maker(o) => OrderForRpc::Maker(MakerOrderForRpc::from(o)),
+            Order::Taker(o) => OrderForRpc::Taker(TakerOrderForRpc::from(o)),
+        }
+    }
+}
+
+#[cfg(target_arch = "wasm32")]
+pub async fn orders_history_by_filter(_ctx: MmArc, _req: Json) -> Result<Response<Vec<u8>>, String> {
+    let res = json!({
+        "error": format!("'orders_history_by_filter' is only supported in native mode yet"),
+    });
+    Response::builder()
+        .status(404)
+        .body(json::to_vec(&res).expect("Serialization failed"))
+        .map_err(|e| ERRL!("{}", e))
+}
+
+/// Returns *all* uuids of swaps, which match the selected filter.
+#[cfg(not(target_arch = "wasm32"))]
+pub async fn orders_history_by_filter(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, String> {
+    use crate::mm2::database::my_orders::select_orders_by_filter;
+
+    let filter: MyOrdersFilter = try_s!(json::from_value(req));
+    let db_result = try_s!(select_orders_by_filter(&ctx.sqlite_connection(), &filter, None));
+
+    let rpc_orders = if filter.include_details {
+        let mut vec = Vec::with_capacity(db_result.result.len());
+        for order in db_result.result.iter() {
+            let uuid = match Uuid::parse_str(order.uuid.as_str()) {
+                Ok(uuid) => uuid,
+                Err(_) => continue,
+            };
+            let order_path = my_orders_history_dir(&ctx).join(order.uuid.clone() + ".json");
+            let content = slurp(&order_path);
+            if let Ok(order) = json::from_slice::<Order>(&content) {
+                vec.push(order);
+                continue;
+            }
+
+            let ordermatch_ctx = try_s!(OrdermatchContext::from_ctx(&ctx));
+            if order.order_type == *"Maker" {
+                let maker_orders = ordermatch_ctx.my_maker_orders.lock().await;
+                if let Some(maker_order) = maker_orders.get(&uuid) {
+                    vec.push(Order::Maker(maker_order.to_owned()));
+                }
+                continue;
+            }
+
+            let taker_orders = ordermatch_ctx.my_taker_orders.lock().await;
+            if let Some(taker_order) = taker_orders.get(&uuid) {
+                vec.push(Order::Taker(taker_order.to_owned()));
+            }
+        }
+        vec
+    } else {
+        vec![]
+    };
+
+    let details = if !rpc_orders.is_empty() {
+        rpc_orders.iter().map(OrderForRpc::from).collect()
+    } else {
+        vec![]
+    };
+
+    let json = json!({
+    "result": {
+        "orders": db_result.result,
+        "details": details,
```

Comment:
If some uuid was filtered out could you please add a warning message in the response too?

Reply:
The `errors` field name can be easily mixed up and considered as an error response in general. Could you rename it to `warnings`, please?